xarray/core/dataarray.py

```diff
@@ -329,7 +329,7 @@ def _replace_maybe_drop_dims(
                                  if set(v.dims) <= allowed_dims)
         return self._replace(variable, coords, name)
 
-    def _replace_indexes(
+    def _overwrite_indexes(
```

Comment:
This wasn't used anywhere in the codebase so I renamed it to be consistent with `Dataset._overwrite_indexes`.